partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -479,17 +479,42 @@ data class StructType(
     val constraints: Set<TupleConstraint> = setOf(),
     override val metas: Map<String, Any> = mapOf(),
 ) : SingleType() {
+
+    public constructor(
+        fields: Map<String, StaticType>,
+        contentClosed: Boolean = false,
+        primaryKeyFields: List<String> = listOf(),
+        constraints: Set<TupleConstraint> = setOf(),
+        metas: Map<String, Any> = mapOf(),
+    ) : this(
+        fields.map { Field(it.key, it.value) },
+        contentClosed,
+        primaryKeyFields,
+        constraints,
+        metas
+    )
+
+    /**
+     * The key-value pair of a StructType, where the key represents the name of the field and the value represents
+     * its [StaticType]. Note: multiple [Field]s within a [StructType] may contain the same [key]. Therefore, when referencing
+     * a field of a [StructType], it is important to gather all fields of the same [key]. To determine the [StaticType]
```

Comment:
> Note: multiple [Field]s within a [StructType] may contain the same [key]. Therefore, when referencing
     * a field of a [StructType], it is important to gather all fields of the same [key]. To determine the [StaticType]

Maybe here we just have to mention that this allows having multiple identical keys different types in the struct as specified by the spec, etc.